sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -338,7 +347,11 @@ public boolean requiresDeduping() {
     @Override
     public void finalizeCheckpoint() throws IOException {
       for (Long sessionId : sessionIds) {
-        channel.basicAck(sessionId, false);
+        try {
+          channel.basicAck(sessionId, false);
+        } catch (IOException e) {
+          channel.basicNack(sessionId, false, true);
```

Comment:
What's the rational here ?

Reply:
I was thinking it would be better to mark the message as unacked if something failed, but after thinking about it a little, it indeed seems a bad idea : if I have the IOException thrown, I won't be able to nack the message, so this is useless.